dnf5/context.cpp

```diff
@@ -210,17 +210,27 @@ void Context::print_info(const char * msg) {
 
 
 void Context::update_repo_load_flags_from_specs(const std::vector<std::string> & pkg_specs) {
+    // TODO(jkolarik): unify this behavior with internal libdnf detection of file specs
     // if a pkg_spec contains '/', it's a path and we need to load filelists
     for (auto & spec : pkg_specs) {
         if (spec.find("/") != std::string::npos) {
-            available_repos_load_flags = available_repos_load_flags | libdnf::repo::LoadFlags::FILELISTS;
-            break;
+            add_repo_load_flags(libdnf::repo::LoadFlags::FILELISTS);
```

Comment:
I am not sure whether `--setopt=metadata_types=filelists` will remove all other types of metadata from configuration.

Reply:
Oh, I didn't realize the `Context` class is in the API. In that case I need to think about it again, because as you wrote there is the ambiguity present. 

So far the `Context` handling was used for commands to put some metadata type to all repo configs which is done there in `Context::load_repos()` using the `RepoQuery`, but it doesn't seem very convenient. Maybe I could create some helper/wrapper method to do this and add it to API, because we can access repo sack through the `Context` API. Then I could drop these `add_repo_load_flags`, `setup_repo_load_flags`, etc. Then really no additional logic dealing with metadata handling will be present.

Reply:
> I am not sure whether `--setopt=metadata_types=filelists` will remove all other types of metadata from configuration.

As it is the `append` option it should not and I think I was also testing this manually from CLI, but I will double-check :slightly_smiling_face:

Reply:
Context part is not yet in API because there is no devel package yet. And thank you verifying `--setopt`.

Reply:
I realized that the appending behavior is currently supported only for CLI bindings. When we want to append a single value to the list from the configuration option I don't see any direct support for this. So this is also why I am handling that in the `Context`. 

Perhaps we could add a support for adding a single value to the OptionStringList instances, but then what about handling the priority?